qa/rolling-upgrade/src/test/java/org/opensearch/knn/bwc/StatsIT.java

```diff
@@ -0,0 +1,64 @@
+/*
+ *  Copyright OpenSearch Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.bwc;
+
+import org.apache.http.util.EntityUtils;
+import org.opensearch.client.Response;
+import org.opensearch.cluster.health.ClusterHealthStatus;
+import org.opensearch.common.xcontent.XContentType;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+
+import static org.opensearch.knn.TestUtils.NODES_BWC_CLUSTER;
+import static org.opensearch.knn.common.KNNConstants.MODEL_INDEX_NAME;
+import static org.opensearch.knn.plugin.stats.StatNames.MODEL_INDEX_STATUS;
+
+public class StatsIT extends AbstractRollingUpgradeTestCase {
+    private static final String MODEL_INDEX_STATUS_NAME = MODEL_INDEX_STATUS.getName();
+
+    // KNN Stats : model_index_status
+    public void testModelIndexHealthMetrics() throws IOException {
+        waitForClusterHealthGreen(NODES_BWC_CLUSTER);
+        switch (getClusterType()) {
+            case OLD:
+                Map<String, Object> statsMap = getModelIndexHealthMetric(MODEL_INDEX_STATUS_NAME);
+                assertNull(statsMap.get(MODEL_INDEX_STATUS_NAME));
+
+                createModelSystemIndex();
```

Comment:
Im a little hesistant on this test. First, the model system index may be being used by other tests so this might interfere with other tests?

We have also ran into flaky tests in the past where we expected one stat value and a race condition produced. Could we change the test to call the stats api at each round and confirm all of the stats are returned (not check value)? I think in the future we can actually check the values.

Reply:
got it, updated